helpers/enable-j5-activate.sh

```diff
@@ -78,5 +78,38 @@ j5activate() {
             return 1
         fi
     fi
+
+  complete -F _j5switch j5switch
+  complete -F _j5setup j5setup
+
+
+}
+_j5setup()
+{
+    local cur=${COMP_WORDS[COMP_CWORD]}
+    COMPREPLY=( $(compgen -W "$(echo "--no-ui --skip-build --help --skip-docs --with-ui --locale --copy-from-custom-staging-dir --no-service-startup --no-db-migration --no-schema-management --admin-password --ignore-installed-apps-requirement --dry-run" | tr " " "\n")" -- $cur ))
+}
+
+_j5switch()
+{
+    local cur=${COMP_WORDS[COMP_CWORD]}
+    if [ $COMP_CWORD -eq 1 ]
+        then
+            COMPREPLY=( $(compgen -W "$(find ~/work/git/ -maxdepth 1 -type d -printf "%f\n")" -- $cur ))
+        else
+            COMPREPLY=( $(compgen -W "$(find ~/j5-homes -maxdepth 1 -type d -printf "%f\n")" -- $cur ))
+    fi
+}
+
+_j5activate()
+{
+    local cur=${COMP_WORDS[COMP_CWORD]}
+    if [ $COMP_CWORD -eq 1 ]
+        then
+            COMPREPLY=( $(compgen -W "$(find ~/j5-environments/ -maxdepth 1 -type d -printf "%f\n" | grep -e "[0-9][0-9].[0-9]")" -- $cur ))
```

Comment:
I don't quite think this is right. This argument is surprisingly not the environment version in j5-environments but it's  actually the work tree suffix (i.e. j5-framework-29.0). So it should probably be something like

```
find ~/work/git/ -maxdepth 1 -name 'j5-framework-*' -type d -printf "%f\n" | cut -d- -f3
```

Reply:
No this is correct - you can't activate something you don't have an environment created for. i.e. j5-framework-23.0 and no ~/j5-envrionments/23.0 so your script would be wrong.  Also the worktree stuff is based on this noth the other way around...

Reply:
Ok scratch that, i have this backwards!

Reply:
fixed